appium/webdriver/webdriver.py

```diff
@@ -799,15 +795,24 @@ def pull_folder(self, path):
         }
         return self.execute(Command.PULL_FOLDER, data)['value']
 
-    def push_file(self, path, base64data):
-        """Puts the data, encoded as Base64, in the file specified as `path`.
+    def push_file(self, destination_path, base64data=None, source_path=None):
+        """Puts the data from the file at `source_path`, encoded as Base64, in the file specified as `path`.
 
+        Specify either `base64data` or `source_path`, if both specified default to `source_path`
         :Args:
-         - path - the path on the device
+         - destination_path - the path on the device
          - base64data - data, encoded as Base64, to be written to the file
+         - source_path - path to file to be loaded on device
         """
+        if source_path is None and base64data is None:
+            raise InvalidArgumentException('Must either present base64 data or a source path on local system')
+
+        if source_path is not None:
+            data = open(source_path, 'rb').read()
```

Comment:
I'd rather prefer using `with` for reading the file

Reply:
agree that a context is a better approach so we make sure to clean up file handlers.

Reply:
You're right, I've gone ahead and made the fix